### ISSUE TITLE
Re-write image stitching using torch/tensorflow

### ISSUE THREAD

**Issue by nrweir** [CosmiQ/solaris#112]
As it stands now, `sol.raster.image.stitch_images` uses numpy to stitch images together. This means that images have to be moved back from the GPU to the CPU to run, and lose any advantage that GPU processing could potentially provide. Particularly since this is almost always done for post-inference images (i.e. `stitch_images()` is called within `sol.nets.infer.Inferer.__call__()`), the objects being merged are likely to often be torch tensors (or could be converted to torch tensors if they're coming from keras). We should therefore implement GPU-based post-processing using `stitch_images()`:

- [ ] re-write `stitch_images()` to use torch tensor operations instead of numpy arrays
- [ ] enable checking for GPU availability and use GPUs if possible
- [ ] enable check to see if data is in a numpy array when it's read in, and if so, convert it to a torch tensor

**Comment by nrweir** [comment]
Resolved with #113
